### ISSUE TITLE
Algorithm Model

### ISSUE THREAD

**Issue by 2735C** [drgn88/ai_age_body_tablet_team4#1]
## AI 알고리즘 진행상황

[데이터 출처](https://www.kaggle.com/code/utkarshsaxenadn/age-gender-detection-multi-model-prediction/notebook?select=UTKFace) 

[데이터 공유 드라이브 링크](https://drive.google.com/drive/folders/1iRhs6a9qfRNIwbJA4BQ22f9b-W7JAy69?usp=drive_link)

### **1차 시도 ➡ 이미지 개수 810** 

```
Training Size  : 648
Validation Size: 81
Testing Size   : 81
```

 
* **Age/Gender Distribution**

<img width="1123" height="500" alt="Image" src="https://github.com/user-attachments/assets/4335bd07-6e46-4c93-8272-ba2f71aeb232" />|<img width="1123" height="500" alt="Image" src="https://github.com/user-attachments/assets/8da4e4f5-fedb-4d6f-9067-658ed39806d5" />|
--|--

<img width="1123" height="500" alt="Image" src="https://github.com/user-attachments/assets/f1612d46-7579-4f39-8410-f7d95926117a" />|<img width="1123" height="500" alt="Image" src="https://github.com/user-attachments/assets/7944df1f-43b5-4b21-8bee-79e5e16aafe4" />|
--|--

* **Sample Image**

<img width="1490" height="956" alt="Image" src="https://github.com/user-attachments/assets/d380a42f-3ebd-43a3-8066-4241a053bcbd" />|
--|

> 1차 학습 결과

  * **Age**

<img width="1489" height="490" alt="Image" src="https://github.com/user-attachments/assets/ede1248e-eea0-4b41-b43e-2901269f86b0" />|
--|

* VGG16

```
Training Loss: 4.9092
Validation Loss: 9.2419
Test Loss: 8.4019
```

* ResNet152V2

```
Training Loss: 2.9664
Validation Loss: 9.9527
Test Loss: 8.4781
```

  * **Gender**

<img width="1490" height="490" alt="Image" src="https://github.com/user-attachments/assets/5b54a849-0f38-43e1-a47e-5b3841794346" />|
--|

* ResNet152V2

```
Training Accuracy:  0.8355, Training Loss: 0.3683
Validation Accuracy: 0.7654, Validation Loss: 0.5371
Test Accuracy: 0.8137, Validation Loss: 0.4491
```

* Xception

```
Training Accuracy:  0.8705, Training Loss: 0.3112
Validation Accuracy: 0.8272, Validation Loss: 0.4126
Test Accuracy: 0.8439, Validation Loss: 0.3831
```

Age Network Learning Curve|Gender Network Learning Curve|
--|--
|<div align = "middle">**VGG16**|<div align = "middle">**ResNet152V2**|
|<img width="563" height="455" alt="Image" src="https://github.com/user-attachments/assets/143d4ecf-e2ef-4966-9e04-41bef1595640" />|<img width="855" height="498" alt="Image" src="https://github.com/user-attachments/assets/b964e1d8-3c95-4020-baed-defb619d42f3" />|
|<div align = "middle">**ResNet152V2**|<div align = "middle">**Xception**|
|<img width="562" height="455" alt="Image" src="https://github.com/user-attachments/assets/8f6e6ba8-1a94-4a7a-b6fc-e6550c6d047d" /> | <img width="855" height="498" alt="Image" src="https://github.com/user-attachments/assets/84319bc3-f9d2-44a2-9ef4-92f9b9bb48b7" />|

* **Result**

VGG16/ResNet152V2|
--|
|<img width="1490" height="956" alt="Image" src="https://github.com/user-attachments/assets/bfb127c8-4684-4547-a44c-f2ef68a747fe" />|
|<div align = "middle">**ResNet152V2**/**Xception**|
|<img width="1490" height="956" alt="Image" src="https://github.com/user-attachments/assets/54101ca6-22e1-4eff-874e-6ba54b12d406" />|

**Comment by 2735C** [comment]
### 2차 시도 ➡ 이미지 개수 10284 (VGG16/ResNet152V2)


```
Training Size  : 8227
Validation Size: 1028
Testing Size   : 1029
```

 
* Age/Gender Distribution

<img width="500" alt="Image" src="https://github.com/user-attachments/assets/7cf44578-8b91-467b-83a2-86161aa08b43" />|<img width="500"  alt="Image" src="https://github.com/user-attachments/assets/fbd1856d-147d-4925-a339-c398441fbb3a" />|
--|--


<img width="500"  alt="Image" src="https://github.com/user-attachments/assets/902d1808-2461-45d4-842b-5ab19662b192" />|<img width="500"  alt="Image" src="https://github.com/user-attachments/assets/a3c9b055-548b-4da6-b339-7db460c87a2b" />
--|--


* Sample Image

<img width="1490" height="956" alt="Image" src="https://github.com/user-attachments/assets/e9373d56-8135-4001-b5b9-d87583641356" />|
--|

<br>

> 1차 학습 결과

  * Age

<img width="1489" height="490" alt="Image" src="https://github.com/user-attachments/assets/0371e0fc-3988-4a24-ad6d-81f1e5a8c966" />|
--|

* VGG16

```
Training Loss: 5.9314
Validation Loss: 7.4992
Test Loss: 7.3039
```

* Xception

```
Training Loss: 4.9485
Validation Loss: 7.6837
Test Loss: 7.4763
```


  * Gender

<img width="1490" height="490" alt="Image" src="https://github.com/user-attachments/assets/53859275-5e10-4c9c-8a96-c346edbb94f6" />|
--|

* ResNet152V2

```
Training Accuracy: 0.8835, Training Loss: 0.2698
Validation Accuracy: 0.8589, Validation Loss: 0.3214
Test Accuracy: 0.8641, Validation Loss:  0.2965
```

* ResNet50V2

```
Training Accuracy: 0.8750, Training Loss: 0.2863
Validation Accuracy: 0.8356, Validation Loss: 0.3607
Test Accuracy: 0.8575, Validation Loss:  0.3288
```

Age Network Learning Curve|Gender Network Learning Curve|
--|--
|<div align = "middle">**VGG16**|<div align = "middle">**ResNet152V2**|
<img width="562" height="455" alt="Image" src="https://github.com/user-attachments/assets/31f6c285-0503-4685-959a-f22057addb81" /> |<img width="855" height="498" alt="Image" src="https://github.com/user-attachments/assets/aba7b80d-6488-467c-815c-053ef7c7dfb3" /> |
|<div align = "middle">**Xception**|<div align = "middle">**ResNet50V2**|
|<img width="562" height="455" alt="Image" src="https://github.com/user-attachments/assets/d02558c9-8160-46ee-b570-e48fbfcb41e2" />|<img width="863" height="498" alt="Image" src="https://github.com/user-attachments/assets/2519f144-92ee-4953-b9f6-f16ca6045a3b" />  |


* Result

VGG16/ResNet152V2|
--|
<img width="1490" height="956" alt="Image" src="https://github.com/user-attachments/assets/a16e2af0-37ad-43b1-9053-2449f66bee1e" />|
|<div align = "middle">**Xception**/**ResNet50V2**|
|  <img width="1490" height="956" alt="Image" src="https://github.com/user-attachments/assets/f4e99462-ff55-46af-b4f2-e733dd2476d6" />|

**Comment by 2735C** [comment]
### ✅ Age Prediction Model (나이 예측 모델)
* Backbone: `VGG16`

* 구조 요약:

<img width="657" height="281" alt="Image" src="https://github.com/user-attachments/assets/29038523-46a6-4a4e-9974-ed203388cab2" />|
--|

* nputLayer (224x224x3)

  * VGG16 (ImageNet pretrained, frozen)

  * Dropout(0.4)

  * Flatten

  * Dense(256, activation="relu")

  * Dense(1) → 나이 출력 (회귀 문제, MAE 사용)

* Loss Function: `Mean Absolute Error (mae)`

* Optimizer: `Adam`

* Checkpoint 파일:
`/content/drive/MyDrive/age_gender/Age-VGG16.keras`

✅ Gender Prediction Model (성별 예측 모델)
* Backbone: `ResNet152V2`

* 구조 요약:

<img width="658" height="263" alt="Image" src="https://github.com/user-attachments/assets/bd29e1c8-4a14-463b-aa94-c313f9931f8c" />|
--|

   * InputLayer (224x224x3)

   * ResNet152V2 (ImageNet pretrained, frozen)

   * Dropout(0.2)

   * GlobalAveragePooling2D

   * Dense(1, activation="sigmoid") → 성별 출력 (이진 분류)

* Loss Function: ` Binary Crossentropy`

* Metric: `Accuracy`

* Optimizer: `Adam`

* Checkpoint 파일:
`/content/drive/MyDrive/age_gender/Gender-ResNet152.keras`

### 왜 이렇게 선택했는가?
* Age Model: 여러 백본 중에서 VGG16이 가장 일관된 학습 곡선을 보였고, train/val loss 간 차이가 적어 신뢰도 있는 회귀 결과를 내는 데 적합했다.

* Gender Model: ResNet152V2는 다른 백본보다 val accuracy가 높고 overfitting이 적었으며, 일반화 성능이 뛰어났다.
